pallets/delegating/src/lib.rs

```diff
@@ -0,0 +1,494 @@
+        let len = target_validators.len();
+        let answer: u64 = total_score / len as u64;
+        let mut remainder: u64 = total_score % len as u64;
+        let mut validators: Vec<(T::AccountId, u64)> = vec![];
+        for v in target_validators {
+            if remainder != 0 {
+                validators.push((v, answer + 1));
+                remainder -= 1;
+            } else {
+                validators.push((v, answer));
+            }
+        }
+        log!(
+            info,
+            "score of: {:?} is {}, delegate to validaors{:?}",
+            delegator,
+            total_score,
+            validators.clone()
+        );
+        validators
+    }
+
+    fn check_and_adjust_delegated_score() {
+        for (delegator, credit_delegate_info) in DelegatedToValidators::<T>::iter() {
+                // check validators in CandidateValidators
+                let target_validators = credit_delegate_info.validators;
+                let mut target_is_changed = false;
+                let candidate_validators = <CandidateValidators<T>>::get().unwrap();
+                let next_target_validators: Vec<_> = target_validators.iter()
+                    .filter(|v|{
+                        if candidate_validators.contains(v) {
+                            true
+                        }else{
+                            target_is_changed = true;
+                            false
+                        }
+                    })
+                    .map(|v| (*v).clone())
+                    .collect();
+
+            // score to low or target_validators not in <CandidateValidators<T>>
+            if T::CreditInterface::pass_threshold(delegator.clone(), 0) == false || next_target_validators.len() == 0 {
+                Self::_undelegate(delegator.clone());
+                <DelegatedToValidators<T>>::remove(delegator.clone());
+            } else {
+                let total_score = T::CreditInterface::get_credit_score(delegator.clone()).unwrap();
+                // score has update or target_validators changed
+                if total_score != credit_delegate_info.score || target_is_changed == true {
+                    Self::_undelegate(delegator.clone());
+
+                    let validators_vec =
+                        Self::cut_credit_score(delegator.clone(), next_target_validators);
+                    Self::_delegate(delegator.clone(), validators_vec);
+
+                    let mut info = <DelegatedToValidators<T>>::take(delegator.clone());
+                    info.score = total_score;
+                    <DelegatedToValidators<T>>::insert(delegator.clone(), info);
+                }
+            }
+        }
+    }
+}
+
+pub trait CreditDelegateInterface<AccountId, B, PB> {
+    fn set_current_era(current_era: EraIndex);
+    fn set_current_era_validators(validators: Vec<AccountId>);
+    fn set_candidate_validators(candidate_validators: Vec<AccountId>);
+
+    /// 获取当前ERA accountid代理的总分
```

Comment:
comment change to: obtain the total delegated score of accountid in current era

Reply:
OK! We will check all comment in code.